DataFormats for Phase-2 L1T-HLT

trtomei opened the pull request:
This PR adds additional infrastructure for the L1T-HLT interface for Phase2.
It is a sister PR to #32137 
We add a couple new DataFormats to allow selections on the new PF-based L1T objects.

We ran `scram b runtests` with success.

We ran `runTheMatrix.py -l limited -i all --ibeos` with success.

trtomei committed:
DataFormats for Phase-2 L1T-HLT

cmsbuild commented:
-code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-32150/19822

- This PR adds an extra 24KB to repository

Code check has found code style and quality issues which could be resolved by applying following patch(s)

- **code-format**:
https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-32150/19822/code-format.patch
e.g. `curl https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-32150/19822/code-format.patch | patch -p1`
You can also run `scram build code-format` to apply code format directly

trtomei committed:
Applied clang-format

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-32150/19832

- This PR adds an extra 24KB to repository

cmsbuild commented:
A new Pull Request was created by  @trtomei (Thiago Tomei) for master.

It involves the following packages:

DataFormats/HLTReco
DataFormats/L1TParticleFlow


@cmsbuild, @kpedro88, @jmduarte, @Martin-Grunewald, @rekovic, @fwyzard can you please review it and eventually sign? Thanks.
@rovere this is something you requested to watch as well.
@silviodonato, @dpiparo, @qliphy you are the release manager for this.

cms-bot commands are listed <a href="http://cms-sw.github.io/cms-bot-cmssw-cmds.html">here</a>

kpedro88 commented on DataFormats/HLTReco/interface/TriggerEventWithRefs.h:
```diff
@@ -300,18 +320,42 @@ namespace trigger {
+    std::pair<size_type, size_type> l1tkmuonSlice(size_type filter) const {
+      const size_type begin(filter == 0 ? 0 : filterObjects_.at(filter - 1).l1tkmuon_);
+      const size_type end(filterObjects_.at(filter).l1tkmuon_);
+      return std::pair<size_type, size_type>(begin, end);
```
slightly easier to use `make_pair` with types inferred, but I recognize this is a pattern already existing in this code

kpedro88 commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
-1

Tested at: 972f0b607854cf8f7e797094179ac2cfbc4fd8ef

CMSSW: CMSSW_11_2_X_2020-11-15-2300
SCRAM_ARCH: slc7_amd64_gcc820
You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-0ceae2/10782/summary.html

I found follow errors while testing this PR

**Failed tests:  Build**

* **Build**:

I found compilation error when building: 

<pre>>> Compiling edm plugin /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_2_X_2020-11-15-2300/src/HLTrigger/HLTcore/plugins/TriggerSummaryAnalyzerAOD.cc
>> Compiling edm plugin /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_2_X_2020-11-15-2300/src/HLTrigger/HLTcore/plugins/TriggerSummaryAnalyzerRAW.cc
>> Compiling edm plugin /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_2_X_2020-11-15-2300/src/HLTrigger/HLTcore/plugins/TriggerSummaryProducerAOD.cc
>> Compiling edm plugin /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_2_X_2020-11-15-2300/src/HLTrigger/HLTcore/plugins/TriggerSummaryProducerRAW.cc
/data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_2_X_2020-11-15-2300/src/HLTrigger/HLTcore/plugins/TriggerSummaryProducerAOD.cc: In member function 'virtual void TriggerSummaryProducerAOD::produce(edm::StreamID, edm::Event&, const edm::EventSetup&) const':
/data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_2_X_2020-11-15-2300/src/HLTrigger/HLTcore/plugins/TriggerSummaryProducerAOD.cc:336:3: error: parse error in template argument list
   fillTriggerObjectCollections<PFJetCollection>(
   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_2_X_2020-11-15-2300/src/HLTrigger/HLTcore/plugins/TriggerSummaryProducerAOD.cc:337:80: error: no matching function for call to 'TriggerSummaryProducerAOD::fillTriggerObjectCollections<<expression error> >(trigger::TriggerObjectCollection&, std::map<edm::ProductID, unsigned int>&, std::vector<std::__cxx11::basic_string<char> >&, trigger::Keys&, edm::Event&, const edm::GetterOfProducts<std::vector<reco::PFJet> >&, TriggerSummaryProducerAOD::InputTagSet&) const'
       toc, offset, tags, keys, iEvent, getPFJetCollection_, collectionTagsEvent);
                                                                                ^
</pre>

cmsbuild commented:
Comparison job queued.

trtomei commented:
Hm, I *did* compile this successfully. Let me see if I understand what is wrong.

trtomei commented:
Okay, turns out that this is more coupled to the HLTrigger/HLTcore than I had understood. Fixing it now.

trtomei committed:
Changes HLTrigger/HLTcore to agree w/ new DataFormats

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-32150/19873

- This PR adds an extra 36KB to repository

cmsbuild commented:
Pull request #32150 was updated. @cmsbuild, @kpedro88, @jmduarte, @Martin-Grunewald, @rekovic, @fwyzard can you please check and sign again.

trtomei commented:
I tested with `checkdeps` now, it will compile! 👍

Martin-Grunewald commented:
Hi @trtomei 
In  HLTrigger/HLTcore/interface/TriggerSummaryProducerAOD.h you only add one of the new l1t types, but everywhere else you add 4 new l1t types. Why is that?

trtomei commented:
> Hi @trtomei
> In HLTrigger/HLTcore/interface/TriggerSummaryProducerAOD|RAW you only add one of the new l1t types, but everywhere else you add 4 new l1t types. Why is that?

I started by doing it indeed, but then noticed that it is not *actually* necessary as far as I understand. It is used only if you want to run those modules, no? 

I think it would be better to revert those changes! What do you say, @Martin-Grunewald ?

Martin-Grunewald commented:
Sorry, but the TriggerSummaryProducerAOD/RAW modules are run in each HLT menu.
The 4 analyzer modules TriggerSummaryAnalyzerAOD|RAW and HLTEventAnalyzerAOD|RAW are expected to report on the full content when run.
So in all 6 of them the 4 added sets of l1t objects should be added.

trtomei commented:
> Sorry, but the TriggerSummaryProducerAOD/RAW modules are run in each HLT menu.
> The 4 analyzer modules TriggerSummaryAnalyzerAOD|RAW and HLTEventAnalyzerAOD|RAW are expected to report on the full content when run.
> So in all 6 of them the 4 added sets of l1t objects should be added.

I never knew that! Thanks for the heads-up, I will add them now.

trtomei committed:
Added missing VectorRefs to classes_def.xml

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-32150/19913

- This PR adds an extra 40KB to repository

cmsbuild commented:
Pull request #32150 was updated. @cmsbuild, @kpedro88, @jmduarte, @Martin-Grunewald, @rekovic, @fwyzard can you please check and sign again.

trtomei committed:
Support for L1 HPS PF Taus (from Talinn group)

cmsbuild commented:
-code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-32150/19966

- This PR adds an extra 20KB to repository

Code check has found code style and quality issues which could be resolved by applying following patch(s)

- **code-checks**:
https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-32150/19966/code-checks.patch
e.g. `curl https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-32150/19966/code-checks.patch | patch -p1`
You can also run `scram build code-checks` to apply code checks directly

trtomei committed:
Support for L1T Phase-2 objects in HLTrigger core

cmsbuild commented:
-code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-32150/20136

- This PR adds an extra 56KB to repository

Code check has found code style and quality issues which could be resolved by applying following patch(s)

- **code-checks**:
https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-32150/20136/code-checks.patch
e.g. `curl https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-32150/20136/code-checks.patch | patch -p1`
You can also run `scram build code-checks` to apply code checks directly

- **code-format**:
https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-32150/20136/code-format.patch
e.g. `curl https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-32150/20136/code-format.patch | patch -p1`
You can also run `scram build code-format` to apply code format directly

trtomei committed:
Add files via upload

trtomei committed:
Add files via upload

trtomei committed:
Add files via upload

trtomei committed:
Add files via upload

cmsbuild commented:
-code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-32150/20144

- This PR adds an extra 56KB to repository

Code check has found code style and quality issues which could be resolved by applying following patch(s)

- **code-checks**:
https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-32150/20144/code-checks.patch
e.g. `curl https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-32150/20144/code-checks.patch | patch -p1`
You can also run `scram build code-checks` to apply code checks directly

- **code-format**:
https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-32150/20144/code-format.patch
e.g. `curl https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-32150/20144/code-format.patch | patch -p1`
You can also run `scram build code-format` to apply code format directly

trtomei committed:
Code check and format

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-32150/20149

- This PR adds an extra 48KB to repository

cmsbuild commented:
Pull request #32150 was updated. @cmsbuild, @kpedro88, @jmduarte, @Martin-Grunewald, @rekovic, @fwyzard can you please check and sign again.

trtomei commented:
I added everything that @Martin-Grunewald asked for, and fixed all problems with dependencies between the new DataFormats and the HLTrigger/HLTcore package. I think this is now ready to review!

fwyzard commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

fwyzard commented on DataFormats/HLTReco/src/classes_def.xml:
```diff
@@ -42,12 +42,16 @@
   </class>
   <class name="trigger::TriggerObjectCollection"/>
-  <class name="trigger::TriggerRefsCollections" ClassVersion="14">
+  <class name="trigger::TriggerRefsCollections" ClassVersion="16">
+   <version ClassVersion="16" checksum="1920377523"/>
+   <version ClassVersion="15" checksum="450679879"/>
```
Here you are adding two new versions - but I assume the intermediate ones (`450679879`, `4165317388`, `2824205283`, `948988526`) are actually never used.
Could you remove them and add only one version to each class ?

For example
```suggestion
  <class name="trigger::TriggerRefsCollections" ClassVersion="15">
   <version ClassVersion="15" checksum="1920377523"/>
```

trtomei replied:
Done!

fwyzard commented on DataFormats/L1TParticleFlow/src/classes.h:
```diff
@@ -6,3 +6,15 @@
 #include "DataFormats/L1TParticleFlow/interface/PFTau.h"
+#include "DataFormats/L1TParticleFlow/interface/HPSPFTau.h"
+#include "DataFormats/L1TParticleFlow/interface/HPSPFTauFwd.h"
+
+namespace DataFormats_L1TParticleFlow {
+  struct dictionary {
+    l1t::HPSPFTau hpspftau;
+    l1t::HPSPFTauCollection hpspftauCollection;
+    edm::Wrapper<l1t::HPSPFTauCollection> hpspftauCWrapper;
+    l1t::HPSPFTauRefVector hpspftaureRefVector;
+    l1t::HPSPFTauVectorRef hpspftauVectorRef;
+  };
+}  // namespace DataFormats_L1TParticleFlow
```
I think this part is not needed any more:
```suggestion
```

fwyzard commented on DataFormats/L1TParticleFlow/interface/HPSPFTau.h:
```diff
@@ -0,0 +1,154 @@
+
+#include <ostream>
+
+// forward declation needed in order to declare HPSPFTauBuilder class as friend (that has access to private data-members)
+class HPSPFTauBuilder;
+
```
I don't actually see any `HPSPFTauBuilder` class in this PR.
Wouldn't it be better to remove the forward declaration and friend declaration for the time being ?
```suggestion
```

trtomei replied:
It may well be! It is my understanding that the Tallinn group has a EDProducer that creates the HPSPFTaus.

fwyzard replied:
Usually we don't need `EDProducer`s to be `friend` of their products - but I can see why in some cases it could make sense.

Anyway, if it will be needed, I would rather add it only when the producer itself is added to CMSSW.

fwyzard commented on DataFormats/L1TParticleFlow/interface/HPSPFTau.h:
```diff
@@ -0,0 +1,154 @@
+    bool passMediumRelIso() const { return passMediumRelIso_; }
+    bool passLooseRelIso() const { return passLooseRelIso_; }
+    bool passVLooseRelIso() const { return passVLooseRelIso_; }
+
+    friend class ::HPSPFTauBuilder;
```
```suggestion
```

cmsbuild commented:
-1

Tested at: d236ecf90393802d61072f4e03aafc78014e2c29

CMSSW: CMSSW_11_3_X_2020-11-27-1100
SCRAM_ARCH: slc7_amd64_gcc900
You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-0ceae2/11121/summary.html

I found follow errors while testing this PR

**Failed tests:  RelVals  AddOn**

* **RelVals**:

When I ran the RelVals I found an error in the following workflows:
136.793 step2
<pre>runTheMatrix-results/136.793_RunDoubleEG2017C+RunDoubleEG2017C+HLTDR2_2017+RECODR2_2017reHLT_skimDoubleEG_Prompt+HARVEST2017/step2_RunDoubleEG2017C+RunDoubleEG2017C+HLTDR2_2017+RECODR2_2017reHLT_skimDoubleEG_Prompt+HARVEST2017.log

</pre>136.731 step2
<pre>runTheMatrix-results/136.731_RunSinglePh2016B+RunSinglePh2016B+HLTDR2_2016+RECODR2_2016reHLT_skimSinglePh_HIPM+HARVESTDR2/step2_RunSinglePh2016B+RunSinglePh2016B+HLTDR2_2016+RECODR2_2016reHLT_skimSinglePh_HIPM+HARVESTDR2.log

</pre>4.53 step2
<pre>runTheMatrix-results/4.53_RunPhoton2012B+RunPhoton2012B+HLTD+RECODR1reHLT+HARVESTDR1reHLT/step2_RunPhoton2012B+RunPhoton2012B+HLTD+RECODR1reHLT+HARVESTDR1reHLT.log

</pre>136.874 step2
<pre>runTheMatrix-results/136.874_RunEGamma2018C+RunEGamma2018C+HLTDR2_2018+RECODR2_2018reHLT_skimEGamma_Offline_L1TEgDQM+HARVEST2018_L1TEgDQM/step2_RunEGamma2018C+RunEGamma2018C+HLTDR2_2018+RECODR2_2018reHLT_skimEGamma_Offline_L1TEgDQM+HARVEST2018_L1TEgDQM.log

</pre>4.22 step2
<pre>runTheMatrix-results/4.22_RunCosmics2011A+RunCosmics2011A+RECOCOSD+ALCACOSD+SKIMCOSD+HARVESTDC/step2_RunCosmics2011A+RunCosmics2011A+RECOCOSD+ALCACOSD+SKIMCOSD+HARVESTDC.log

</pre>140.53 step2
<pre>runTheMatrix-results/140.53_RunHI2011+RunHI2011+RECOHID11+HARVESTDHI/step2_RunHI2011+RunHI2011+RECOHID11+HARVESTDHI.log

</pre>8.0 step2
<pre>runTheMatrix-results/8.0_BeamHalo+BeamHalo+DIGICOS+RECOCOS+ALCABH+HARVESTCOS/step2_BeamHalo+BeamHalo+DIGICOS+RECOCOS+ALCABH+HARVESTCOS.log

</pre>140.56 step2
<pre>runTheMatrix-results/140.56_RunHI2018+RunHI2018+RECOHID18+HARVESTDHI18/step2_RunHI2018+RunHI2018+RECOHID18+HARVESTDHI18.log

</pre>1306.0 step2
<pre>runTheMatrix-results/1306.0_SingleMuPt1_UP15+SingleMuPt1_UP15+DIGIUP15+RECOUP15+HARVESTUP15/step2_SingleMuPt1_UP15+SingleMuPt1_UP15+DIGIUP15+RECOUP15+HARVESTUP15.log

</pre>7.3 step2
<pre>runTheMatrix-results/7.3_CosmicsSPLoose_UP18+CosmicsSPLoose_UP18+DIGICOS_UP18+RECOCOS_UP18+ALCACOS_UP18+HARVESTCOS_UP18/step2_CosmicsSPLoose_UP18+CosmicsSPLoose_UP18+DIGICOS_UP18+RECOCOS_UP18+ALCACOS_UP18+HARVESTCOS_UP18.log

</pre>312.0 step2
<pre>runTheMatrix-results/312.0_Pyquen_ZeemumuJets_pt10_2760GeV_2021+Pyquen_ZeemumuJets_pt10_2760GeV_2021+DIGIHI2021MIX+RECOHI2021MIX+HARVESTHI2021PPRECO/step2_Pyquen_ZeemumuJets_pt10_2760GeV_2021+Pyquen_ZeemumuJets_pt10_2760GeV_2021+DIGIHI2021MIX+RECOHI2021MIX+HARVESTHI2021PPRECO.log

</pre>1000.0 step2
<pre>runTheMatrix-results/1000.0_RunMinBias2011A+RunMinBias2011A+TIER0+SKIMD+HARVESTDfst2+ALCASPLIT/step2_RunMinBias2011A+RunMinBias2011A+TIER0+SKIMD+HARVESTDfst2+ALCASPLIT.log

</pre>1330.0 step2
<pre>runTheMatrix-results/1330.0_ZMM_13+ZMM_13+DIGIUP15+RECOUP15_L1TMuDQM+HARVESTUP15_L1TMuDQM+NANOUP15/step2_ZMM_13+ZMM_13+DIGIUP15+RECOUP15_L1TMuDQM+HARVESTUP15_L1TMuDQM+NANOUP15.log

</pre>9.0 step2
<pre>runTheMatrix-results/9.0_Higgs200ChargedTaus+Higgs200ChargedTaus+DIGI+RECO+HARVEST/step2_Higgs200ChargedTaus+Higgs200ChargedTaus+DIGI+RECO+HARVEST.log

</pre>1001.0 step2
<pre>runTheMatrix-results/1001.0_RunMinBias2011A+RunMinBias2011A+TIER0EXP+ALCAEXP+ALCAHARVDSIPIXELCALRUN1+ALCAHARVD1+ALCAHARVD2+ALCAHARVD3+ALCAHARVD4+ALCAHARVD5/step2_RunMinBias2011A+RunMinBias2011A+TIER0EXP+ALCAEXP+ALCAHARVDSIPIXELCALRUN1+ALCAHARVD1+ALCAHARVD2+ALCAHARVD3+ALCAHARVD4+ALCAHARVD5.log

</pre>10042.0 step2
<pre>runTheMatrix-results/10042.0_ZMM_13+2017+ZMM_13TeV_TuneCUETP8M1_GenSim+Digi+RecoFakeHLT+HARVESTFakeHLT+ALCA+Nano/step2_ZMM_13+2017+ZMM_13TeV_TuneCUETP8M1_GenSim+Digi+RecoFakeHLT+HARVESTFakeHLT+ALCA+Nano.log

</pre>25.0 step2
<pre>runTheMatrix-results/25.0_TTbar+TTbar+DIGI+RECOAlCaCalo+HARVEST+ALCATT/step2_TTbar+TTbar+DIGI+RECOAlCaCalo+HARVEST+ALCATT.log

</pre>11624.911 step1
<pre>runTheMatrix-results/11624.911_TTbar_13+2021_DD4hep+TTbar_13TeV_TuneCUETP8M1_GenSim+Digi+Reco+HARVEST+ALCA/step1_TTbar_13+2021_DD4hep+TTbar_13TeV_TuneCUETP8M1_GenSim+Digi+Reco+HARVEST+ALCA.log

</pre>25202.0 step2
<pre>runTheMatrix-results/25202.0_TTbar_13+TTbar_13+DIGIUP15_PU25+RECOUP15_PU25+HARVESTUP15_PU25+NANOUP15_PU25/step2_TTbar_13+TTbar_13+DIGIUP15_PU25+RECOUP15_PU25+HARVESTUP15_PU25+NANOUP15_PU25.log

</pre>10024.0 step2
<pre>runTheMatrix-results/10024.0_TTbar_13+2017+TTbar_13TeV_TuneCUETP8M1_GenSim+Digi+RecoFakeHLT+HARVESTFakeHLT+ALCA+Nano/step2_TTbar_13+2017+TTbar_13TeV_TuneCUETP8M1_GenSim+Digi+RecoFakeHLT+HARVESTFakeHLT+ALCA+Nano.log

</pre>11634.0 step2
<pre>runTheMatrix-results/11634.0_TTbar_14TeV+2021+TTbar_14TeV_TuneCP5_GenSim+Digi+Reco+HARVEST+ALCA/step2_TTbar_14TeV+2021+TTbar_14TeV_TuneCP5_GenSim+Digi+Reco+HARVEST+ALCA.log

</pre>12434.0 step2
<pre>runTheMatrix-results/12434.0_TTbar_14TeV+2023+TTbar_14TeV_TuneCP5_GenSim+Digi+Reco+HARVEST+ALCA/step2_TTbar_14TeV+2023+TTbar_14TeV_TuneCP5_GenSim+Digi+Reco+HARVEST+ALCA.log

</pre>10824.0 step2
<pre>runTheMatrix-results/10824.0_TTbar_13+2018+TTbar_13TeV_TuneCUETP8M1_GenSim+Digi+RecoFakeHLT+HARVESTFakeHLT+ALCA+Nano/step2_TTbar_13+2018+TTbar_13TeV_TuneCUETP8M1_GenSim+Digi+RecoFakeHLT+HARVESTFakeHLT+ALCA+Nano.log

</pre>10224.0 step2
<pre>runTheMatrix-results/10224.0_TTbar_13+2017PU+TTbar_13TeV_TuneCUETP8M1_GenSim+DigiPU+RecoFakeHLTPU+HARVESTFakeHLTPU+Nano/step2_TTbar_13+2017PU+TTbar_13TeV_TuneCUETP8M1_GenSim+DigiPU+RecoFakeHLTPU+HARVESTFakeHLTPU+Nano.log

</pre>23234.0 step2
<pre>runTheMatrix-results/23234.0_TTbar_14TeV+2026D49+TTbar_14TeV_TuneCP5_GenSimHLBeamSpot14+DigiTrigger+RecoGlobal+HARVESTGlobal/step2_TTbar_14TeV+2026D49+TTbar_14TeV_TuneCP5_GenSimHLBeamSpot14+DigiTrigger+RecoGlobal+HARVESTGlobal.log

</pre>28234.0 step2
<pre>runTheMatrix-results/28234.0_TTbar_14TeV+2026D60+TTbar_14TeV_TuneCP5_GenSimHLBeamSpot14+DigiTrigger+RecoGlobal+HARVESTGlobal/step2_TTbar_14TeV+2026D60+TTbar_14TeV_TuneCP5_GenSimHLBeamSpot14+DigiTrigger+RecoGlobal+HARVESTGlobal.log

</pre>250202.181 step3
<pre>runTheMatrix-results/250202.181_TTbar_13UP18+TTbar_13UP18+PREMIXUP18_PU25+DIGIPRMXLOCALUP18_PU25+RECOPRMXUP18_PU25+HARVESTUP18_PU25/step3_TTbar_13UP18+TTbar_13UP18+PREMIXUP18_PU25+DIGIPRMXLOCALUP18_PU25+RECOPRMXUP18_PU25+HARVESTUP18_PU25.log

</pre>23434.999 step3
<pre>runTheMatrix-results/23434.999_TTbar_14TeV+2026D49PU_PMXS1S2PR+TTbar_14TeV_TuneCP5_GenSimHLBeamSpot14+PREMIX_PremixHLBeamSpot14PU+DigiTriggerPU+RecoGlobalPU+HARVESTGlobalPU/step3_TTbar_14TeV+2026D49PU_PMXS1S2PR+TTbar_14TeV_TuneCP5_GenSimHLBeamSpot14+PREMIX_PremixHLBeamSpot14PU+DigiTriggerPU+RecoGlobalPU+HARVESTGlobalPU.log

</pre>

* **AddOn**:

I found errors in the following addon tests:

cmsRun /cvmfs/cms-ib.cern.ch/week0/slc7_amd64_gcc900/cms/cmssw/CMSSW_11_3_X_2020-11-27-1100/src/HLTrigger/Configuration/test/OnLine_HLT_Fake2.py realData=True  globalTag=@ inputFiles=@  : FAILED - time: date Sat Nov 28 01:01:49 2020-date Sat Nov 28 01:00:23 2020 s - exit: 18176
cmsDriver.py RelVal -s HLT:Fake2,RAW2DIGI,L1Reco,RECO --data --scenario=pp -n 10 --conditions auto:run2_data_Fake2 --relval 9000,50 --datatier "RAW-HLT-RECO" --eventcontent FEVTDEBUGHLT --customise=HLTrigger/Configuration/CustomConfigs.L1THLT --era Run2_2016 --processName=HLTRECO --filein file:RelVal_Raw_Fake2_DATA.root --fileout file:RelVal_Raw_Fake2_DATA_HLT_RECO.root : FAILED - time: date Sat Nov 28 01:01:49 2020-date Sat Nov 28 01:00:23 2020 s - exit: 18176
cmsRun /cvmfs/cms-ib.cern.ch/week0/slc7_amd64_gcc900/cms/cmssw/CMSSW_11_3_X_2020-11-27-1100/src/HLTrigger/Configuration/test/OnLine_HLT_PRef.py realData=True  globalTag=@ inputFiles=@  : FAILED - time: date Sat Nov 28 01:02:19 2020-date Sat Nov 28 01:00:26 2020 s - exit: 18176
cmsDriver.py RelVal -s HLT:PRef,RAW2DIGI,L1Reco,RECO --data --scenario=pp -n 10 --conditions auto:run3_data_PRef --relval 9000,50 --datatier "RAW-HLT-RECO" --eventcontent FEVTDEBUGHLT --customise=HLTrigger/Configuration/CustomConfigs.L1THLT --era Run3 --processName=HLTRECO --filein file:RelVal_Raw_PRef_DATA.root --fileout file:RelVal_Raw_PRef_DATA_HLT_RECO.root : FAILED - time: date Sat Nov 28 01:02:19 2020-date Sat Nov 28 01:00:26 2020 s - exit: 18176
cmsRun /cvmfs/cms-ib.cern.ch/week0/slc7_amd64_gcc900/cms/cmssw/CMSSW_11_3_X_2020-11-27-1100/src/HLTrigger/Configuration/test/OnLine_HLT_HIon.py realData=True  globalTag=@ inputFiles=@  : FAILED - time: date Sat Nov 28 01:02:21 2020-date Sat Nov 28 01:00:29 2020 s - exit: 18176
cmsDriver.py RelVal -s HLT:HIon,RAW2DIGI,L1Reco,RECO --data --scenario=pp -n 10 --conditions auto:run3_data_HIon --relval 9000,50 --datatier "RAW-HLT-RECO" --eventcontent FEVTDEBUGHLT --customise=HLTrigger/Configuration/CustomConfigs.L1THLT --era Run3_pp_on_PbPb --processName=HLTRECO --filein file:RelVal_Raw_HIon_DATA.root --fileout file:RelVal_Raw_HIon_DATA_HLT_RECO.root : FAILED - time: date Sat Nov 28 01:02:21 2020-date Sat Nov 28 01:00:29 2020 s - exit: 18176
cmsRun /cvmfs/cms-ib.cern.ch/week0/slc7_amd64_gcc900/cms/cmssw/CMSSW_11_3_X_2020-11-27-1100/src/HLTrigger/Configuration/test/OnLine_HLT_GRun.py realData=False globalTag=@ inputFiles=@  : FAILED - time: date Sat Nov 28 01:06:04 2020-date Sat Nov 28 01:00:31 2020 s - exit: 18176
cmsDriver.py RelVal -s HLT:GRun,RAW2DIGI,L1Reco,RECO --mc --scenario=pp -n 10 --conditions auto:run3_mc_GRun --relval 9000,50 --datatier "RAW-HLT-RECO" --eventcontent FEVTDEBUGHLT --customise=HLTrigger/Configuration/CustomConfigs.L1THLT --era Run3 --processName=HLTRECO --filein file:RelVal_Raw_GRun_MC.root --fileout file:RelVal_Raw_GRun_MC_HLT_RECO.root : FAILED - time: date Sat Nov 28 01:06:04 2020-date Sat Nov 28 01:00:31 2020 s - exit: 18176
cmsRun /cvmfs/cms-ib.cern.ch/week0/slc7_amd64_gcc900/cms/cmssw/CMSSW_11_3_X_2020-11-27-1100/src/HLTrigger/Configuration/test/OnLine_HLT_PRef.py realData=False globalTag=@ inputFiles=@  : FAILED - time: date Sat Nov 28 01:05:33 2020-date Sat Nov 28 01:00:37 2020 s - exit: 18176
cmsDriver.py RelVal -s HLT:PRef,RAW2DIGI,L1Reco,RECO --mc --scenario=pp -n 10 --conditions auto:run3_mc_PRef --relval 9000,50 --datatier "RAW-HLT-RECO" --eventcontent FEVTDEBUGHLT --customise=HLTrigger/Configuration/CustomConfigs.L1THLT --era Run3 --processName=HLTRECO --filein file:RelVal_Raw_PRef_MC.root --fileout file:RelVal_Raw_PRef_MC_HLT_RECO.root : FAILED - time: date Sat Nov 28 01:05:33 2020-date Sat Nov 28 01:00:37 2020 s - exit: 18176
cmsRun /cvmfs/cms-ib.cern.ch/week0/slc7_amd64_gcc900/cms/cmssw/CMSSW_11_3_X_2020-11-27-1100/src/HLTrigger/Configuration/test/OnLine_HLT_PIon.py realData=True  globalTag=@ inputFiles=@  : FAILED - time: date Sat Nov 28 01:03:46 2020-date Sat Nov 28 01:00:45 2020 s - exit: 18176
cmsDriver.py RelVal -s HLT:PIon,RAW2DIGI,L1Reco,RECO --data --scenario=pp -n 10 --conditions auto:run3_data_PIon --relval 9000,50 --datatier "RAW-HLT-RECO" --eventcontent FEVTDEBUGHLT --customise=HLTrigger/Configuration/CustomConfigs.L1THLT --era Run3 --processName=HLTRECO --filein file:RelVal_Raw_PIon_DATA.root --fileout file:RelVal_Raw_PIon_DATA_HLT_RECO.root : FAILED - time: date Sat Nov 28 01:03:46 2020-date Sat Nov 28 01:00:45 2020 s - exit: 18176
cmsRun /cvmfs/cms-ib.cern.ch/week0/slc7_amd64_gcc900/cms/cmssw/CMSSW_11_3_X_2020-11-27-1100/src/HLTrigger/Configuration/test/OnLine_HLT_Fake.py realData=False globalTag=@ inputFiles=@  : FAILED - time: date Sat Nov 28 01:05:33 2020-date Sat Nov 28 01:00:50 2020 s - exit: 18176
cmsDriver.py RelVal -s HLT:Fake,RAW2DIGI,L1Reco,RECO --mc --scenario=pp -n 10 --conditions auto:run1_mc_Fake --relval 9000,50 --datatier "RAW-HLT-RECO" --eventcontent FEVTDEBUGHLT --customise=HLTrigger/Configuration/CustomConfigs.L1THLT --processName=HLTRECO --filein file:RelVal_Raw_Fake_MC.root --fileout file:RelVal_Raw_Fake_MC_HLT_RECO.root : FAILED - time: date Sat Nov 28 01:05:33 2020-date Sat Nov 28 01:00:50 2020 s - exit: 18176
cmsRun /cvmfs/cms-ib.cern.ch/week0/slc7_amd64_gcc900/cms/cmssw/CMSSW_11_3_X_2020-11-27-1100/src/HLTrigger/Configuration/test/OnLine_HLT_HIon.py realData=False globalTag=@ inputFiles=@  : FAILED - time: date Sat Nov 28 01:05:51 2020-date Sat Nov 28 01:00:51 2020 s - exit: 18176
cmsDriver.py RelVal -s HLT:HIon,RAW2DIGI,L1Reco,RECO --mc --scenario=pp -n 10 --conditions auto:run3_mc_HIon --relval 9000,50 --datatier "RAW-HLT-RECO" --eventcontent FEVTDEBUGHLT --customise=HLTrigger/Configuration/CustomConfigs.L1THLT --era Run3_pp_on_PbPb --processName=HLTRECO --filein file:RelVal_Raw_HIon_MC.root --fileout file:RelVal_Raw_HIon_MC_HLT_RECO.root : FAILED - time: date Sat Nov 28 01:05:51 2020-date Sat Nov 28 01:00:51 2020 s - exit: 18176
cmsRun /cvmfs/cms-ib.cern.ch/week0/slc7_amd64_gcc900/cms/cmssw/CMSSW_11_3_X_2020-11-27-1100/src/HLTrigger/Configuration/test/OnLine_HLT_Fake2.py realData=False globalTag=@ inputFiles=@  : FAILED - time: date Sat Nov 28 01:05:37 2020-date Sat Nov 28 01:00:53 2020 s - exit: 18176
cmsDriver.py RelVal -s HLT:Fake2,RAW2DIGI,L1Reco,RECO --mc --scenario=pp -n 10 --conditions auto:run2_mc_Fake2 --relval 9000,50 --datatier "RAW-HLT-RECO" --eventcontent FEVTDEBUGHLT --customise=HLTrigger/Configuration/CustomConfigs.L1THLT --era Run2_2016 --processName=HLTRECO --filein file:RelVal_Raw_Fake2_MC.root --fileout file:RelVal_Raw_Fake2_MC_HLT_RECO.root : FAILED - time: date Sat Nov 28 01:05:37 2020-date Sat Nov 28 01:00:53 2020 s - exit: 18176
cmsRun /cvmfs/cms-ib.cern.ch/week0/slc7_amd64_gcc900/cms/cmssw/CMSSW_11_3_X_2020-11-27-1100/src/HLTrigger/Configuration/test/OnLine_HLT_Fake1.py realData=False globalTag=@ inputFiles=@  : FAILED - time: date Sat Nov 28 01:05:36 2020-date Sat Nov 28 01:00:54 2020 s - exit: 18176
cmsDriver.py RelVal -s HLT:Fake1,RAW2DIGI,L1Reco,RECO --mc --scenario=pp -n 10 --conditions auto:run2_mc_Fake1 --relval 9000,50 --datatier "RAW-HLT-RECO" --eventcontent FEVTDEBUGHLT --customise=HLTrigger/Configuration/CustomConfigs.L1THLT --era Run2_25ns --processName=HLTRECO --filein file:RelVal_Raw_Fake1_MC.root --fileout file:RelVal_Raw_Fake1_MC_HLT_RECO.root : FAILED - time: date Sat Nov 28 01:05:36 2020-date Sat Nov 28 01:00:54 2020 s - exit: 18176
cmsRun /cvmfs/cms-ib.cern.ch/week0/slc7_amd64_gcc900/cms/cmssw/CMSSW_11_3_X_2020-11-27-1100/src/HLTrigger/Configuration/test/OnLine_HLT_Fake.py realData=True  globalTag=@ inputFiles=@  : FAILED - time: date Sat Nov 28 01:03:27 2020-date Sat Nov 28 01:00:59 2020 s - exit: 18176
cmsDriver.py RelVal -s HLT:Fake,RAW2DIGI,L1Reco,RECO --data --scenario=pp -n 10 --conditions auto:run1_data_Fake --relval 9000,50 --datatier "RAW-HLT-RECO" --eventcontent FEVTDEBUGHLT --customise=HLTrigger/Configuration/CustomConfigs.L1THLT --processName=HLTRECO --filein file:RelVal_Raw_Fake_DATA.root --fileout file:RelVal_Raw_Fake_DATA_HLT_RECO.root : FAILED - time: date Sat Nov 28 01:03:27 2020-date Sat Nov 28 01:00:59 2020 s - exit: 18176
cmsRun /cvmfs/cms-ib.cern.ch/week0/slc7_amd64_gcc900/cms/cmssw/CMSSW_11_3_X_2020-11-27-1100/src/HLTrigger/Configuration/test/OnLine_HLT_GRun.py realData=True  globalTag=@ inputFiles=@  : FAILED - time: date Sat Nov 28 01:05:46 2020-date Sat Nov 28 01:01:12 2020 s - exit: 18176
cmsDriver.py RelVal -s HLT:GRun,RAW2DIGI,L1Reco,RECO --data --scenario=pp -n 10 --conditions auto:run3_data_GRun --relval 9000,50 --datatier "RAW-HLT-RECO" --eventcontent FEVTDEBUGHLT --customise=HLTrigger/Configuration/CustomConfigs.L1THLT --era Run3 --processName=HLTRECO --filein file:RelVal_Raw_GRun_DATA.root --fileout file:RelVal_Raw_GRun_DATA_HLT_RECO.root : FAILED - time: date Sat Nov 28 01:05:46 2020-date Sat Nov 28 01:01:12 2020 s - exit: 18176
cmsRun /cvmfs/cms-ib.cern.ch/week0/slc7_amd64_gcc900/cms/cmssw/CMSSW_11_3_X_2020-11-27-1100/src/HLTrigger/Configuration/test/OnLine_HLT_Fake1.py realData=True  globalTag=@ inputFiles=@  : FAILED - time: date Sat Nov 28 01:04:43 2020-date Sat Nov 28 01:01:17 2020 s - exit: 18176
cmsDriver.py RelVal -s HLT:Fake1,RAW2DIGI,L1Reco,RECO --data --scenario=pp -n 10 --conditions auto:run2_data_Fake1 --relval 9000,50 --datatier "RAW-HLT-RECO" --eventcontent FEVTDEBUGHLT --customise=HLTrigger/Configuration/CustomConfigs.L1THLT --era Run2_25ns --processName=HLTRECO --filein file:RelVal_Raw_Fake1_DATA.root --fileout file:RelVal_Raw_Fake1_DATA_HLT_RECO.root : FAILED - time: date Sat Nov 28 01:04:43 2020-date Sat Nov 28 01:01:17 2020 s - exit: 18176
cmsRun /cvmfs/cms-ib.cern.ch/week0/slc7_amd64_gcc900/cms/cmssw/CMSSW_11_3_X_2020-11-27-1100/src/HLTrigger/Configuration/test/OnLine_HLT_PIon.py realData=False globalTag=@ inputFiles=@  : FAILED - time: date Sat Nov 28 01:06:55 2020-date Sat Nov 28 01:01:50 2020 s - exit: 18176
cmsDriver.py RelVal -s HLT:PIon,RAW2DIGI,L1Reco,RECO --mc --scenario=pp -n 10 --conditions auto:run3_mc_PIon --relval 9000,50 --datatier "RAW-HLT-RECO" --eventcontent FEVTDEBUGHLT --customise=HLTrigger/Configuration/CustomConfigs.L1THLT --era Run3 --processName=HLTRECO --filein file:RelVal_Raw_PIon_MC.root --fileout file:RelVal_Raw_PIon_MC_HLT_RECO.root : FAILED - time: date Sat Nov 28 01:06:55 2020-date Sat Nov 28 01:01:50 2020 s - exit: 18176

cmsbuild commented:
Comparison not run due to runTheMatrix errors (RelVals and Igprof tests were also skipped)

fwyzard commented:
Looks like some dictionaries are still missing ?
```c++
vector<edm::Ref<vector<l1t::HPSPFTau>,l1t::HPSPFTau,edm::refhelper::FindUsingAdvance<vector<l1t::HPSPFTau>,l1t::HPSPFTau> > >
vector<edm::Ref<vector<l1t::PFTau>,l1t::PFTau,edm::refhelper::FindUsingAdvance<vector<l1t::PFTau>,l1t::PFTau> > >
vector<edm::Ref<vector<l1t::PFTrack>,l1t::PFTrack,edm::refhelper::FindUsingAdvance<vector<l1t::PFTrack>,l1t::PFTrack> > >
```

missirol commented on HLTrigger/HLTcore/plugins/HLTEventAnalyzerRAW.cc:
```diff
@@ -349,6 +370,79 @@ void HLTEventAnalyzerRAW::analyzeTrigger(const edm::Event& iEvent,
+      triggerEventWithRefsHandle_->getObjects(filterIndex, l1tpftauIds_, l1tpftauRefs_);
+      const unsigned int nL1TPFTaus(l1tpftauIds_.size());
+      if (nL1TPFTaus > 0) {
+        LogVerbatim("HLTEventAnalyzerRAW") << "   PFJets: " << nL1TPFTaus << "  - the objects: # id pt" << endl;
```
```suggestion
        LogVerbatim("HLTEventAnalyzerRAW") << "   L1TPFTaus: " << nL1TPFTaus << "  - the objects: # id pt" << endl;
```

missirol commented on HLTrigger/HLTcore/plugins/HLTEventAnalyzerRAW.cc:
```diff
@@ -349,6 +370,79 @@ void HLTEventAnalyzerRAW::analyzeTrigger(const edm::Event& iEvent,
+      triggerEventWithRefsHandle_->getObjects(filterIndex, l1thpspftauIds_, l1thpspftauRefs_);
+      const unsigned int nL1THPSPFTaus(l1thpspftauIds_.size());
+      if (nL1THPSPFTaus > 0) {
+        LogVerbatim("HLTEventAnalyzerRAW") << "   PFJets: " << nL1THPSPFTaus << "  - the objects: # id pt" << endl;
```
```suggestion
        LogVerbatim("HLTEventAnalyzerRAW") << "   L1THPSPFTaus: " << nL1THPSPFTaus << "  - the objects: # id pt" << endl;
```

missirol commented on HLTrigger/HLTcore/plugins/HLTEventAnalyzerRAW.cc:
```diff
@@ -349,6 +370,79 @@ void HLTEventAnalyzerRAW::analyzeTrigger(const edm::Event& iEvent,
+      triggerEventWithRefsHandle_->getObjects(filterIndex, l1tpftrackIds_, l1tpftrackRefs_);
+      const unsigned int nL1TPFTracks(l1tpftrackIds_.size());
+      if (nL1TPFTracks > 0) {
+        LogVerbatim("HLTEventAnalyzerRAW") << "   PFJets: " << nL1TPFTracks << "  - the objects: # id pt" << endl;
```
```suggestion
        LogVerbatim("HLTEventAnalyzerRAW") << "   L1TPFTracks: " << nL1TPFTracks << "  - the objects: # id pt" << endl;
```

missirol commented on HLTrigger/HLTcore/plugins/TriggerSummaryAnalyzerRAW.cc:
```diff
@@ -157,6 +157,35 @@ void TriggerSummaryAnalyzerRAW::analyze(edm::StreamID, const edm::Event& iEvent,
+      if (nL1TPFTau > 0)
+        LogVerbatim("TriggerSummaryAnalyzerRAW") << " L1TPFTau: " << nL1TPFTau;
+
+      const unsigned int nL1THPSPFTau(handle->l1tetsumSlice(iFO).second - handle->l1tetsumSlice(iFO).first);
```
```suggestion
      const unsigned int nL1THPSPFTau(handle->l1thpspftauSize(iFO).second - handle->l1thpspftauSize(iFO).first);
```

missirol commented on HLTrigger/HLTcore/plugins/TriggerSummaryAnalyzerRAW.cc:
```diff
@@ -157,6 +157,35 @@ void TriggerSummaryAnalyzerRAW::analyze(edm::StreamID, const edm::Event& iEvent,
+      if (nL1THPSPFTau > 0)
+        LogVerbatim("TriggerSummaryAnalyzerRAW") << " L1THPSPFTau: " << nL1THPSPFTau;
+
+      const unsigned int nL1TPFTrack(handle->l1tetsumSlice(iFO).second - handle->l1tetsumSlice(iFO).first);
```
```suggestion
      const unsigned int nL1TPFTrack(handle->l1tpftrackSize(iFO).second - handle->l1tpftrackSize(iFO).first);
```

trtomei replied:
All addressed, but it was "Slice", not "Size" -- that's why I didn't accept the Suggestion here. Thanks for the review!

kpedro88 commented on DataFormats/L1TParticleFlow/interface/HPSPFTau.h:
```diff
@@ -0,0 +1,154 @@
+/// print to stream
+std::ostream& operator<<(std::ostream& os, const l1t::HPSPFTau& l1PFTau);
+
+//void printPFCand(ostream& os, const l1t::PFCandidate& l1PFCand, const l1t::VertexRef& primaryVertex);
```
delete commented-out code

trtomei replied:
Done!

kpedro88 commented on DataFormats/L1TParticleFlow/interface/HPSPFTau.h:
```diff
@@ -0,0 +1,154 @@
+std::ostream& operator<<(std::ostream& os, const l1t::HPSPFTau& l1PFTau);
+
+//void printPFCand(ostream& os, const l1t::PFCandidate& l1PFCand, const l1t::VertexRef& primaryVertex);
+void printPFCand(ostream& os, const l1t::PFCandidate& l1PFCand, const l1t::TkPrimaryVertexRef& primaryVertex);
```
maybe these functions should be in the `l1t` namespace (rather than the global namespace)?

trtomei replied:
Done!

kpedro88 commented on DataFormats/L1TParticleFlow/src/HPSPFTau.cc:
```diff
@@ -0,0 +1,87 @@
+  } else if (l1PFTau.isPFJetSeeded()) {
+    os << " PFJet";
+  } else
+    assert(0);
```
a clearer error message (exception) could be provided here

trtomei replied:
It will now throw a cms::Exception with more information.

kpedro88 commented on DataFormats/L1TParticleFlow/src/HPSPFTau.cc:
```diff
@@ -0,0 +1,87 @@
+  return os;
+}
+
+//void printPFCand(ostream& os, const l1t::PFCandidate& l1PFCand, const l1t::VertexRef& primaryVertex)
```
delete commented-out code

trtomei replied:
Done!

trtomei committed:
Update classes_def.xml

Removing the intermediate class versions in classes_def.xml

trtomei committed:
Update DataFormats/L1TParticleFlow/src/classes.h

Removed the additional dictionary of collections, Refs, etc. for HPSPFTau

Co-authored-by: Andrea Bocci <andrea.bocci@cern.ch>

trtomei committed:
Update DataFormats/L1TParticleFlow/interface/HPSPFTau.h

Co-authored-by: Andrea Bocci <andrea.bocci@cern.ch>

trtomei committed:
Update HLTrigger/HLTcore/plugins/HLTEventAnalyzerRAW.cc

Correct printout.

Co-authored-by: Marino Missiroli <m.missiroli@cern.ch>

trtomei committed:
Update HLTrigger/HLTcore/plugins/HLTEventAnalyzerRAW.cc

Correct printout.

Co-authored-by: Marino Missiroli <m.missiroli@cern.ch>

trtomei committed:
Update HLTrigger/HLTcore/plugins/HLTEventAnalyzerRAW.cc

Correct printout.

Co-authored-by: Marino Missiroli <m.missiroli@cern.ch>

trtomei committed:
Update HPSPFTau.h

Suggestions by Kevin Pedro on functions that print PF candidates and deleting commented-out code.

trtomei committed:
Update HPSPFTau.cc

Deleting commented-out code.

cmsbuild commented:
-code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-32150/20182

ERROR: Build errors found during clang-tidy run.
```
Suppressed 582 warnings (577 in non-user code, 4 NOLINT, 1 with check filters).
--
DataFormats/L1TParticleFlow/interface/HPSPFTau.h:82:20: error: no class named 'HPSPFTauBuilder' in the global namespace [clang-diagnostic-error]
    friend class ::HPSPFTauBuilder;
                   ^
Suppressed 1214 warnings (1213 in non-user code, 1 with check filters).
--
DataFormats/L1TParticleFlow/interface/HPSPFTau.h:82:20: error: no class named 'HPSPFTauBuilder' in the global namespace [clang-diagnostic-error]
    friend class ::HPSPFTauBuilder;
                   ^
Suppressed 1378 warnings (1377 in non-user code, 1 with check filters).
--
DataFormats/L1TParticleFlow/interface/HPSPFTau.h:82:20: error: no class named 'HPSPFTauBuilder' in the global namespace [clang-diagnostic-error]
    friend class ::HPSPFTauBuilder;
                   ^
Suppressed 1335 warnings (1328 in non-user code, 6 NOLINT, 1 with check filters).
--
DataFormats/L1TParticleFlow/interface/HPSPFTau.h:82:20: error: no class named 'HPSPFTauBuilder' in the global namespace [clang-diagnostic-error]
    friend class ::HPSPFTauBuilder;
                   ^
Suppressed 1918 warnings (1911 in non-user code, 6 NOLINT, 1 with check filters).
--
gmake: *** [config/SCRAM/GMake/Makefile.coderules:128: code-checks] Error 2
gmake: *** [There are compilation/build errors. Please see the detail log above.] Error 2
```

trtomei committed:
Better error handling in HPSPFTau, plus other fixes

trtomei committed:
Correct printouts in the EventAnalyzers

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-32150/20184

- This PR adds an extra 120KB to repository

cmsbuild commented:
Pull request #32150 was updated. @cmsbuild, @kpedro88, @jmduarte, @Martin-Grunewald, @rekovic, @fwyzard can you please check and sign again.

Martin-Grunewald commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: 8fe8a28c8495025ea8cbe835962bf8365a9034fa
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-0ceae2/11193/summary.html
CMSSW: CMSSW_11_3_X_2020-11-30-1100
SCRAM_ARCH: slc7_amd64_gcc900

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-0ceae2/11193/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 4 differences found in the comparisons
 - DQMHistoTests: Total files compared: 35
 - DQMHistoTests: Total histograms compared: 2529593
 - DQMHistoTests: Total failures: 2
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2529569
 - DQMHistoTests: Total skipped: 22
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 34 files compared)
 - Checked 148 log files, 34 edm output root files, 35 DQM output files

kpedro88 commented:
+upgrade

Martin-Grunewald commented:
+1

trtomei commented:
Dear @rekovic , @cbotta ,  please check this PR. Notice that this is *not* the CondFormats PR, this is the the *DataFormats* one. We need this one to allow selections on the new PF-based L1T objects, independent of any scaling functions.

rekovic commented:
+1

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next master IBs (tests are also fine). This pull request will now be reviewed by the release team before it's merged. @silviodonato, @dpiparo, @qliphy (and backports should be raised in the release meeting by the corresponding L2)

silviodonato commented:
+1
